src/common/telemetry.ts

```diff
@@ -267,19 +257,17 @@ export module Telemetry {
                 let userType: string = TelemetryUtils.telemetrySettings.userType;
 
                 if (userType === undefined) {
-                    if (process.env[TelemetryUtils.INTERNAL_USER_ENV_VAR]) {
-                        userType = TelemetryUtils.USERTYPE_INTERNAL;
-                    } else if (os.platform() === "win32") {
-                        let domain: string = process.env.USERDNSDOMAIN;
-                        domain = domain ? domain.toLowerCase().substring(domain.length - TelemetryUtils.INTERNAL_DOMAIN_SUFFIX.length) : null;
-                        userType = domain === TelemetryUtils.INTERNAL_DOMAIN_SUFFIX ? TelemetryUtils.USERTYPE_INTERNAL : TelemetryUtils.USERTYPE_EXTERNAL;
-                    } else {
-                        userType = TelemetryUtils.USERTYPE_EXTERNAL;
-                    }
-
-                    TelemetryUtils.telemetrySettings.userType = userType;
+                if (process.env[TelemetryUtils.INTERNAL_USER_ENV_VAR]) {
+                    userType = TelemetryUtils.USERTYPE_INTERNAL;
+                } else {
+                    let domain: string = process.env.USERDNSDOMAIN;

```

Comment:
I verified that it is not defined on OSX to maintain existing code logic. However, I think we can also have internal users on other platforms, so assuming it is defined on other platform configurations, we should be fine.